018-inertiajs.md

```diff
@@ -0,0 +1,81 @@
+
+This is where Inertia JS would excel at providing the ideal solution; it would allow us to enable a fully client-side rendered frontend and provides the glue to connect it to our existing server-side Laravel PHP code.
+
+The updates to connect the pieces are minimal and gets the team up and running to develop our administrative frontend utilizing React components and Tailwind.
+
+## Proposal
+
+Implement the use of [Inertia JS](https://inertiajs.com/) in Northstar for our administrative interface.
+
+The overall aim is to reduce maintenance for our team, particularly when adding or updating new features to our administrative interfaces.
+
+Inertia will allow us to use the best of both worlds; a client-side rendered interface with reusable React components and server-side business logic utilizing the robust and familiar Laravel framework without worrying about additional code to pass data to the React components.
+
+There will be an initial ramp up to build out the core components for our administrative interfaces. However, we can build on the experience we have obtained from component-izing our user facing frontend. We also already have some components from our activity service that have been built using React, and we could easily update those to receive data props from Inertia. Once we have an established baseline of admin components most of the admin pages will be easier to build out and/or update.
+
+The following is a proposed approach for implementing Inertia JS in Northstar:
+
+- Add Inertia JS to Northstar and set it up so it works with our existing code.
+- Implement a new main layout for the admin interface.
+- Consolidate all routes into Inertia routes that will pass data as props.
+- Build out the interface using React components that utilize the data passed via the Inertia routes.
+
+## Drawbacks
+
+For the most part there do not appear to be any significant drawbacks with implementing Inertia JS. It seems like the right "happy middleground" that would let us maintain most of our existing Laravel backend work, yet provide the flexibility to use component-based interface elements built in React.
+
+One potential drawback is that the team would be required to update and create admin interface elements; especially given how specialized some of our needs can be at times. However, I imagine that after building a number of initial admin interface elements in React as components, the level of effort would drop as components can be re-used.
+
+Another potential drawback is that Inertia just has not been around for that long. The [Inertia adaptor for Laravel](https://github.com/inertiajs/inertia-laravel) is currently tagged at version `v0.4.1` and the [Inertia adaptor for React](https://github.com/inertiajs/inertia/blob/master/packages/inertia-react/package.json) is currently at version `v0.5.12`.
```

Comment:
Do we know about other companies/organizations that are using Inertia in production? It would make me feel a lot more comfortable picking up a newer tool if, say, some of our non-profit friends were already using it!

Reply:
I added some additional info regarding this!